docs/assemblies/dataplane_performance_tuning_large_scale.adoc

```diff
@@ -0,0 +1,725 @@
+        ansibleHost: 192.168.122.125
+    # ... compute-26 through compute-49
+----
+
+*Advantages:*
+
+* *Increased parallelism*: Multiple ansible-runner pods execute simultaneously
+  for each NodeSet.
+* Better failure isolation - one NodeSet's failure doesn't block others
+* Easier troubleshooting and isolation of issues
+* Can deploy groups incrementally or independently
+* Lower memory per ansible-runner pod
+
+*Disadvantages:*
+
+* More NodeSet CRs to manage and monitor
+* Potential for configuration drift between NodeSets
+* Higher OpenShift overhead - multiple ansible-runner pods
+* More complex deployment orchestration
+
+=== Strategy 3: Role-Based NodeSets
+
+Group nodes by their role or function rather than arbitrarily.
+
+.Role-Based NodeSets Example
+[source,yaml]
+----
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneNodeSet
+metadata:
+  name: compute-standard
+spec:
+  nodes:
+    compute-0:
+      hostName: compute-0
+      ansible:
+        ansibleHost: 192.168.122.100
+    # ... standard compute nodes
+---
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneNodeSet
+metadata:
+  name: compute-gpu
+spec:
+  nodeTemplate:
+    ansible:
+      ansibleVars:
+        edpm_nova_pci_passthrough_whitelist:
+          - '{"vendor_id": "10de", "product_id": "1b38"}'
+  nodes:
+    compute-gpu-0:
+      hostName: compute-gpu-0
+      ansible:
+        ansibleHost: 192.168.122.210
+    # ... GPU compute nodes
+----
+
+*Advantages:*
+
+* Different configurations for different node types
+* Clear organizational structure
+* Natural parallelism across different roles
+* Easier capacity planning and management
+
+*Disadvantages:*
+
+* May not maximize parallelism if roles have different node counts
+* Configuration must be carefully managed across NodeSets
+
+Shares similar advantages and disadvantages as using multiple smaller NodeSets.
+
+== Parallel Execution Patterns
+
+Understanding how the openstack-operator executes deployments is key to optimization.
+
+=== NodeSet-Level Parallelism
+
+When you create an `OpenStackDataPlaneDeployment` with multiple NodeSets, the operator starts them sequentially but they execute in parallel:
+
+.Multiple NodeSets Deployment
+[source,yaml]
+----
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneDeployment
+metadata:
+  name: edpm-deployment
+spec:
+  nodeSets:
+    - compute-group-1
+    - compute-group-2
+    - compute-group-3
+    - compute-group-4
+----
+
+*Execution flow:*
+
+1. Operator starts deployment for `compute-group-1` → ansible-runner pod launches
+2. Operator starts deployment for `compute-group-2` → ansible-runner pod launches
+3. Operator starts deployment for `compute-group-3` → ansible-runner pod launches
+4. Operator starts deployment for `compute-group-4` → ansible-runner pod launches
+5. All four ansible-runner pods execute in parallel
+
+This means *4 separate Ansible executions run simultaneously*, each processing
```

Comment:
Maybe it's mentioned elsewhere I missed, we can probably mention about AnsibleJobNodeSelector for deployments that can distribute them across worker nodes.